lib/Echidna/Campaign.hs

```diff
@@ -174,9 +184,27 @@ execTxOptC t = do
   res <- execTxWith vmExcept (usingCoverage $ pointCoverage (hasLens . coverage)) t
   hasLens . coverage %= unionWith union og
   grew <- (== LT) . comparing coveragePoints og <$> use (hasLens . coverage)
-  when grew $ hasLens . genDict %= gaddCalls ([t ^. call] ^.. traverse . _SolCall)
+  --when grew $ hasLens . genDict %= gaddCalls ([t ^. call] ^.. traverse . _SolCall)
+  when grew $ hasLens . newCoverage .= True
   return res
 
+-- | Given save a list of transactions in the corpus, discarding reverts
+addToCorpus :: (MonadState s m, Has Campaign s) => [(Tx, VMResult)] -> m ()
+addToCorpus res = let rtxs = map fst res --(filter (\(_, vm) -> classifyRes vm /= ResRevert) res)
+                    in unless (null rtxs) $ hasLens . corpus %= (rtxs:)
+
+-- | Generate a new sequences of transactions, either using the corpus or with randomly created transactions
+randseq :: ( MonadCatch m, MonadRandom m, MonadReader x m, MonadState y m
+           , Has GenDict y, Has TxConf x, Has TestConf x, Has CampaignConf x, Has Campaign y)
+        => Int -> Map Addr Contract -> World -> m [Tx]
+randseq ql o w = do ca <- use hasLens
+                    let txs = ca ^. corpus  
+                        p   = ca ^. ncallseqs 
+                    if length txs > p then -- Replay the transactions in the corpus, if we are executing the first iterations
+                      return $ txs !! p
+                    else                        -- Randomly generate new transactions 
+                      replicateM ql (evalStateT (genTxM o) (w, ca ^. genDict))
```

Comment:
let's review whether or not we need this `evalStateT` now

Reply:
Good question. Any thoughts?